src/sinks/prometheus/exporter.rs

```diff
@@ -271,29 +274,35 @@ impl StreamSink for PrometheusExporter {
             let item = event.into_metric();
             let mut metrics = self.metrics.write().unwrap();
 
+            // sets need to be expired from time to time
+            // because otherwise they could grow infinitelly
+            let now = Utc::now().timestamp();
+            let interval = now - metrics.last_flush_timestamp;
+            if interval > self.config.flush_period_secs as i64 {
+                metrics.last_flush_timestamp = now;
+
+                let mut map = IndexMap::new();
+                for (MetricEntry(mut metric), is_incremental_set) in metrics.map.drain(..) {
```

Comment:
Not sure that `drain` is the best way to reset keys, but do not see any other options.

Reply:
I think you could also do this with:
```rust
let mut map = std::mem::replace(&mut metrics.map, IndexMap::new());
for (MetricEntry(mut metric), is_incremental_set) in &mut map {
    if is_incremental_set {
        metric.reset()
    }
}
```

Reply:
Never mind my last comment, I missed that you assign back to `metrics.map`.

I think you're right that this needs a `drain`. The only other way would be to modify the keys in place, and that doesn't work well with iterating. I'd approach it like this:
```rust
let map = metrics.map
    .drain()
    .map(|(MetricEntry(mut metric), is_incremental_set))| {
        if is_incremental_set {
            metric.reset();
        }
        (MetricEntry(metric), is_incremental_set)
    })
    .collect();
metrics.map = map;
```